infrastructure/sandbox/PreProvisioner/lambda/Dockerfile

```diff
@@ -1,5 +1,31 @@
-FROM golang:1.18-alpine AS builder
-RUN apk update && apk add --no-cache git curl openssl unzip
+FROM rust:latest AS builder
+
+ARG transporter_url=https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa/ra/resources/download/public/Transporter__Linux/bin
+
+RUN cargo install apple-codesign \
+  && curl -sSf $transporter_url -o transporter_install.sh \
+  && sh transporter_install.sh --target transporter --accept --noexec
+
+FROM golang:1.18.4-bullseye
+
+RUN apt-get update \
+  && dpkg --add-architecture i386 \
+  && apt update \
+  && apt install -y --no-install-recommends ca-certificates cpio libxml2 wine wine32 libgtk-3-0 \
+  && rm -rf /var/lib/apt/lists/* 
+
+# copy macOS dependencies
+COPY --from=fleetdm/bomutils:latest /usr/bin/mkbom /usr/local/bin/xar /usr/bin/
+COPY --from=fleetdm/bomutils:latest /usr/local/lib /usr/local/lib/
+COPY --from=builder /transporter/itms /usr/local/
+COPY --from=builder /usr/local/cargo/bin/rcodesign /usr/local/bin
```

Comment:
we can publish `fleetdm/fleetctl` and you can copy from there instead of using a builder, just lmk.

Reply:
this was copied from a dockerfile that builds fleetctl. There was some talk with @lucasmrod about this and we should probably have a wider discussion about it. Basically the goal is that when I terraform apply, the fleet code currently checked out in local git is the code that runs. This means importing dockerfiles can be difficult since we would need to publish fleetctl, then import it somewhere.

I think later on we can pretty much completely detach the infra release stuff from fleet proper release stuff once the interfaces mature more, but for right now since we don't have a mature interface between fleetctl and cloud requirements being able to iterate more quickly is more valuable.

Reply:
sounds good to me!

Reply:
been thinking about this, and perhaps a better way would be to publish an image for `rcodesign`, which will take the bulk of the image build time here, and also is a bit problematic to build on M1.

I created an issue https://github.com/fleetdm/fleet/issues/7052